webapp/sdf/src/components/list-component/list-component.spec.js

```diff
@@ -0,0 +1,141 @@
+import { mount } from '@vue/test-utils'
+import ListComponent from 'components/list-component/list-component.vue'
+import Item from 'classes/item.js'
+import ItemComponent from 'components/item-component/item-component.vue'
+
+describe('list-component', () => {
+    describe('item list', () => {
+        describe('empty', () => {
+            it ('shows message "The list is empty!"', async () => {
+                let component = mount(ListComponent, {
+                    data() { return {itemList: []}}
+                })
+                const h2Element = component.find('.list-empty')
+                expect(h2Element.text()).toEqual("The list is empty!")
+            })
+        })
+
+        describe('not empty', () => {
+            let component
+
+            beforeEach(() => {
+                component = mount(ListComponent, {
+                    data() { return {
+                        itemList: [
+                            new Item(1, "Eintrag1", 5),
+                            new Item(2, "Eintrag2", 2),
+                            new Item(3, "Eintrag3", -2)
+                        ]
+                    }}
+                })
+            })
+
+            describe('renders ItemComponent for each item', () => {
+                it('renders 3 components', async () => {
+                    const itemComponents = component.findAllComponents(ItemComponent)
+                    expect(itemComponents.length).toBe(3)
+                })
+            })
+
+            describe('receives remove event for item 1', () => {
+                it('removes item 1', async () => {
+                    component.findComponent(ItemComponent).vm.$emit('removeEvent', {
+                        id:1, title:"Eintrag1", votes:5
+                    })
+                    await component.vm.$nextTick()
+                    expect(component.vm.$data.itemList.some(item =>
+                        item.id == 1 &&
+                        item.title =="Eintrag1" &&
+                        item.votes == 5
+                    )).toBe(false)
```

Comment:
This is too error-prone. I would suggest `Arrange - Assert - Act- Assert`:
```js
expect(component.findAllComponents).toHaveLength(3)
await component.find('button.remove').trigger('click')
expect(component.findAllComponents).toHaveLength(2)
```

Reply:
By the way, I just found: https://stackoverflow.com/questions/1021007/should-it-be-arrange-assert-act-assert

I like the way of calling it: Arrange - *Assume* - Act - Assert